### DISCUSSION TITLE
Prisma: MongoDB error: An error occurred during DNS resolution: request timed out

### DISCUSSION THREAD

**Discussion by soorajlegend** [prisma/prisma#19673] [Q&A] [answered]
I created my models in prisma/schema.prisma and I'm trying to push them to my MongoDB using npx prisma db push, however, its always returns this error: 
Error: MongoDB error
An error occurred during DNS resolution: request timed out
   0: schema_core::commands::schema_push::Calculate `from`
             at schema-engine\core\src\commands\schema_push.rs:43
   1: schema_core::state::SchemaPush
             at schema-engine\core\src\state.rs:433

**Comment by ludralph** [comment]
Hi @soorajlegend 👋 

This looks like a problem with your connection to your MongoDB database. How are you running MongoDB? Can you connect to this instance with other tools?

**Reply by soorajlegend** [reply]
I actually never use it, so this is my first time, I tried using the command palette but is also not working

**Reply by ludralph** [reply]
Can you connect to your MongoDB instance using a tool like [MongoDB Compass](https://www.mongodb.com/products/compass)?

**Reply by soorajlegend** [reply]
well I never tried it, But I can give it a try now

**Reply by ludralph** [reply]
Let me know if it works for you.

**Reply by soorajlegend** [reply]
noted

**Reply by soorajlegend** [reply]
Nothing works for me...and I ain't using any VPN, I even tried changing my DNS server address

**Reply by ludralph** [reply]
Are you using antivirus software? If yes, you may want to disable it. Something is definitely blocking it.

**Reply by soorajlegend** [reply, marked as answer]
Hello Sir, Finally the problem was from my Windows operating system and I just shut down the system at night and on in in the morning to continue debugging, fortunately, the command just worked. Thank you so much for your support.

**Comment by imad-oi** [comment]
please can someone help ,i have the same error

**Comment by AsharSi** [comment]
I am facing the same issue. I had used Prisma to connect mongodb from my nextjs app. When I had used command "npx prisma db push" it shows that error. I don't know what is causing that issue. Also I had seen the MongoDB is not connecting with the database.

**Reply by soorajlegend** [reply]
Well check your network thoroughly, if possible restart your system, make sure you're not connected to a proxy or VPN. if it persists try using another network. I had such issue more than once and I noticed the problem is mainly from my phone's network, when I connect to my office network everything works without changing anything, so try multiple ways, hopefully you'll win the war!!!

**Reply by Taha-mlaiki** [reply]
guys can you help me , It's been 2 days still facing this problem :
Error: MongoDB error
Kind: An error occurred during DNS resolution: proto error: io error: A socket operation was attempted to an unreachable network. (os error 10051), labels: {}
   0: schema_core::commands::schema_push::Calculate `from`
             at schema-engine\core\src\commands\schema_push.rs:27
   1: schema_core::state::SchemaPush
             at schema-engine\core\src\state.rs:504

**Comment by soorajlegend** [comment]
In my case the problem was my network, I switched to another network and it
worked fine. Make sure you ain’t using a proxy or vpn connection
